opcua/server/address_space.py

```diff
@@ -260,6 +285,36 @@ def _add_reference(self, addref, user):
         self._aspace[addref.SourceNodeId].references.append(rdesc)
         return ua.StatusCode()
 
+    def delete_references(self, refs, user=User.Admin):
+        result = []
+        for ref in refs:
+            result.append(self._delete_reference(ref, user))
+        return result
+
+    def _delete_reference(self, item, user):
+        if item.SourceNodeId not in self._aspace:
+            return ua.StatusCode(ua.StatusCodes.BadSourceNodeIdInvalid)
+        if item.TargetNodeId not in self._aspace:
+            return ua.StatusCode(ua.StatusCodes.BadTargetNodeIdInvalid)
+        if not user == User.Admin:
+            return ua.StatusCode(ua.StatusCodes.BadUserAccessDenied)
+
+        for rdesc in self._aspace[item.SourceNodeId].references:
+            if rdesc.NodeId is item.TargetNodeId:
+                if rdesc.RefrenceTypeId != item.RefrenceTypeId:
+                    return ua.StatusCode(ua.StatusCode.BadReferenceTypeInvalid)
+                if rdesc.IsForward == item.IsForward or item.DeleteBidirectional:
+                    self._aspace[item.SourceNodeId].references.remove(rdesc)
+
+        for rdesc in self._aspace[item.TargetNodeId].references:
+            if rdesc.NodeId is item.SourceNodeId:
+                if rdesc.RefrenceTypeId != item.RefrenceTypeId:
+                    return ua.StatusCode(ua.StatusCode.BadReferenceTypeInvalid)
+                if rdesc.IsForward == item.IsForward or item.DeleteBidirectional:
+                    self._aspace[item.SourceNodeId].references.remove(rdesc)
+
+        return ua.StatusCode()
+
```

Comment:
This is something I am not sure myself if I implemented it good. I am not sure to understand completely the references by OPC UA.... I implemented it because kind a "stayed in a way" :)